### PR TITLE
[CELEBORN-1487] CongestionController support control traffic by user/worker traffic speed

### PR DESCRIPTION
**Pull request by leixm**
### What changes were proposed in this pull request?
Introduce support control traffic by user/worker traffic speed.

### Why are the changes needed?
Currently, Celeborn only supports quota management based on disk file bytes/count, and this quota management cannot cope with sudden increases in traffic, which will cause corrupt to the cluster.

### Does this PR introduce _any_ user-facing change?
Yes.

### How was this patch tested?
UTs.

### CODE REVIEW

File: worker/src/main/java/org/apache/celeborn/service/deploy/worker/congestcontrol/CongestionController.java [hunk 1]
```diff
@@ -217,15 +268,22 @@ private void removeInactiveUsers() {
   protected void checkCongestion() {
     try {
       long pendingConsume = getTotalPendingBytes();
-      if (pendingConsume < lowWatermark) {
+      long workerProduceSpeed = consumedBufferStatusHub.avgBytesPerSec();
```
Comment (AngersZhuuuu) [inline worker/src/main/java/org/apache/celeborn/service/deploy/worker/congestcontrol/CongestionController.java:271]: Maybe we should use produce bytes?

**Comment by leixm** [comment]
https://docs.google.com/document/d/15ALP-Wy0CktO7PX39oXpZjIi0llFxos0HQ53KnYFq5E/edit?usp=sharing   cc @RexXiong  @AngersZhuuuu  Can you help review design doc?

**Comment by mridulm** [comment]
Can you open up access to the doc @leixm  ? Thanks

**Comment by leixm** [comment]
> Can you open up access to the doc @leixm ? Thanks

Sorry, I have updated the doc access, can you access now?

**Comment by leixm** [comment]
Flaky test.

File: worker/src/main/java/org/apache/celeborn/service/deploy/worker/congestcontrol/CongestionController.java (hunk 1 above)
Comment (leixm) [reply worker/src/main/java/org/apache/celeborn/service/deploy/worker/congestcontrol/CongestionController.java]: Sure, I have updated.

**Pull request closed without merging**